Source/Core/VideoBackends/OGL/BoundingBox.cpp

```diff
@@ -2,69 +2,167 @@
 // Licensed under GPLv2+
 // Refer to the license.txt file included.
 
+#include <algorithm>
+#include <array>
 #include <cstring>
 
 #include "Common/GL/GLUtil.h"
 
 #include "VideoBackends/OGL/BoundingBox.h"
+#include "VideoBackends/OGL/FramebufferManager.h"
 
 #include "VideoCommon/DriverDetails.h"
 #include "VideoCommon/VideoConfig.h"
 
 static GLuint s_bbox_buffer_id;
+static GLuint s_pbo;
+
+static std::array<int, 4> s_stencil_bounds{{0, 0, 0, 0}};
+static bool s_stencil_updated;
+static bool s_stencil_cleared;
+
+static int s_target_width;
+static int s_target_height;
 
 namespace OGL
 {
-void BoundingBox::Init()
+void BoundingBox::SetTargetSizeChanged(int target_width, int target_height)
+{
+  if (g_ActiveConfig.BBoxUseFragmentShaderImplementation())
+    return;
+
+  s_target_width = target_width;
```

Comment:
I don't understand. Are you asking, why not call g_renderer->GetTargetWidth/Height() in the rest of the code?

Since SetTargetSizeChanged needs to be called anyways, this is an easy opportunity to reduce decoupling/header-dependency on RenderBase.